modules/blockapis/src/BaseHttpClient.ts

```diff
@@ -29,7 +28,13 @@ export interface HttpClient {
 }
 
 export function mapSeries<T, U>(arr: T[], f: (v: T, i: number) => Promise<U>): Promise<U[]> {
-  return Bluebird.mapSeries(arr, f);
+  const results: U[] = [];
+  return (async () => {
+    for (const [index, value] of arr.entries()) {
+      results.push(await f(value, index));
+    }
+    return results;
+  })();
```

Comment:
why is is IIFE?

Reply:
good catch, no need for IIFE. turned it into async function